src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/AnsiTerminalTestProgressFrame.cs

```diff
@@ -228,12 +296,57 @@ public void Render(AnsiTerminalTestProgressFrame previousFrame, AnsiTerminal ter
             terminal.AppendLine();
         }
 
-        // clear no longer used lines
-        if (i < previousFrame.ProgressCount)
+        // We rendered more lines in previous frame. Clear them.
+        if (previousFrame.RenderedLines != null && i < previousFrame.RenderedLines.Count)
         {
             terminal.Append($"{AnsiCodes.CSI}{AnsiCodes.EraseInDisplay}");
         }
     }
 
-    public void Clear() => ProgressCount = 0;
+    private List<object> GenerateLinesToRender(TestProgressState?[] progress)
+    {
+        var linesToRender = new List<object>(progress.Length);
+
+        // Note: We want to render the list of active tests, but this can easily fill up the full screen.
+        // As such, we should balance the number of active tests shown per project.
+        // We do this by distributing the remaining lines for each projects.
+        var progressItems = progress.OfType<TestProgressState>().ToList();
+        int linesToDistribute = (int)(Height * 0.7) - 1 - progressItems.Count;
+        var detailItems = new IEnumerable<TestDetailState>[progressItems.Count];
+        var sortedItemsI = Enumerable.Range(0, progressItems.Count).OrderBy(i => progressItems[i].TestNodeResultsState?.Count ?? 0).ToList();
```

Comment:
What do you mean by the `I` suffix?

Reply:
that the list contains the indices of sorted items. could have used `Indices` instead

Reply:
Oh ok yeah please do, let's avoid abbreviations.

Reply:
changed